### PR TITLE
[Website] Add ADBC 0.5.0 release post

### PR DESCRIPTION
**Pull request by lidavidm**

### CODE REVIEW

**Review by kou — APPROVED** [review]
+1

File: _posts/2023-06-22-adbc-0.5.0-release.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+A milestone has been created to explore [asynchronous APIs][adbc-api-1.2.0], which have been requested for a while now.
+
+## Getting Involved
+
+We welcome questions and contributions from all interested.  Issues
+can be filed on [GitHub][4], and questions can be directed to GitHub
+or the [Arrow mailing lists][5].
+
+[1]: https://github.com/apache/arrow-adbc/milestone/6
```
Comment (raulcd) [inline _posts/2023-06-22-adbc-0.5.0-release.md:83]: Should we link to the closed tab to see the issues?
```suggestion
[1]: https://github.com/apache/arrow-adbc/milestone/6?closed=1
```

**Review by raulcd — APPROVED** [review]
LGTM just two minor nits

File: _posts/2023-06-22-adbc-0.5.0-release.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+0.5.0.  The **API specification** is versioned separately and is
+at version 1.0.0.
+
+The release notes below are not exhaustive and only expose selected
+highlights of the release. Many other bugfixes and improvements have
+been made: we refer you to the [complete changelog][3].
+
+## Release Highlights
+
+Experimental C♯ and Rust codebases were added to the source tree.  No packages are released for them yet.
```
[the remark this answers is not in the document]
Comment (lidavidm) [reply _posts/2023-06-22-adbc-0.5.0-release.md]: Ah, it turns out the sharp is only used for logos and such:

> By convention, a [number sign](https://en.wikipedia.org/wiki/Number_sign) is used for the second character in normal text; in artistic representations, sometimes a true [sharp sign](https://en.wikipedia.org/wiki/%E2%99%AF) is used: C♯. However the [ECMA](https://en.wikipedia.org/wiki/Ecma_International) 334 standard states: "The name C# is written as the LATIN CAPITAL LETTER C (U+0043) followed by the NUMBER SIGN # (U+0023)."

I guess Unicode is still too hard...

**Pull request merged**